### PR TITLE
docs: ads

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

Ads framework docs.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Comment by kaltura-hooks** [comment]
Hi @dan-ziv,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: docs/ads.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+The Kaltura TV Platform Player supports the addition of 3rd party monetization solutions via its plugin system. To allow easy integration and to support playback and analytics using any 3rd party solution, the player defines an advertizing framework using APIs and a standard event scheme.
+
+Any plugin that adds support for ads is expected to follow the controller, timeline, and events scheme.
+This document details the ads controller, which allows controlling ad-related operations, the ad timeline (which explains how a typical order of events looks like), and the ad events and their associated payload.
+
+## Ads Controller
+
+To query the player regarding the ad's state in the current playback, you can use the ads controller by referencing `player.ads`.
+The ads controller is responsible for tracking and communicating with the active ads plugin, as well as for collecting and saving the current ads state and data.
```
Comment (AnnaAleksandrowicz) [inline docs/ads.md]: I'm not clear here if in "the current ads state and data" you mean a single ad or plural? In any case it should probably have a possessive ' added: "collecting and saving the current ad's state and data" or "collecting and saving the current ads' state and data".
Clear?

File: docs/ads.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+## Ad Events Timeline
+
+During ad playback, there are events that get triggered to indicate the current ad state or state changes.
+Some events are life cycle indicators and will always get triggered, while others are reactive to specific actions -- such as the user clicking the ad -- and may or may not be triggered.  
+
+In the following diagram you can observe the events timeline from the point that an ad break starts to the point that the ad break ends.
+
+![ad-events-timeline](images/ad-events-timeline.jpg)
+
+If you want to know when playback has started/ended -- whether the playback includes ads or not -- use the following events:
```
Comment (AnnaAleksandrowicz) [inline docs/ads.md]: You used the term "aware" - I changed to "know" unless you want to use the term "be made aware" - up to you!

File: docs/ads.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+- `PLAYBACK_START` - Triggered on the first play request for playback. If a pre-roll ad is about to play, this event will be triggered before the pre-roll. If no pre-roll ad is about to play, this event will be triggered before any other content event including `FIRST_PLAY`.
+
+- `PLAYBACK_ENDED` - Triggered when playback is ended. If a post-roll ad is about to play, this event will be triggered after post-roll. If no post-roll ad is about to play, this event will be triggered after any other content event including `ENDED`.
```
Comment (AnnaAleksandrowicz) [inline docs/ads.md]: I change the first part of both pre-roll and post-roll to "is about to play" - verify that you are ok with this change.

File: docs/ads.md [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+> ### <a name="AD_BREAK_END"></a>AD_BREAK_END
+>
+> Fires when an ad break is finished.
+
+#
+
+> ### <a name="AD_FIRST_QUARTILE"></a>AD_FIRST_QUARTILE
+>
+> Fires when the ad playhead crosses first the quartile.
```
Comment (AnnaAleksandrowicz) [inline docs/ads.md]: Just verifying that "ad playhead" is a term that's use in the industry; I saw that BrightCove also use it but just checking...

File: docs/ads.md [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+- `id` **[string][32]** Ad ID.
+- `options` **PKAdOptions** - The ad data options.
+
+### id
+
+Returns **[string][32]** - The ad ID.
+
+### contentType
+
+Returns **[string][32]** - The ad content type.
```
Comment (AnnaAleksandrowicz) [inline docs/ads.md]: This description originally said "The ad ID" but that's wrong, right? just verifying...

File: docs/ads.md (hunk 1 above)
Comment (OrenMe) [reply docs/ads.md]: @AnnaAleksandrowicz I believe it should describe any ad's state, which is currently being played.
I changed it to `To query the player regarding an ad's state in the current playback`

File: docs/ads.md (hunk 4 above)
Comment (OrenMe) [reply docs/ads.md]: @AnnaAleksandrowicz yes, this is a coined term

**Pull request merged**